File: src/algorithms/array-sign.js
/**
 * Given an integer array nums, returns 1 if the product of the array is
 * positive, returns -1 if the product is negative, or returns 0 if the product
 * is equal to 0;
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number} - Returns a number signifying that the product of the array
 *     is either positive (1), negative (-1), or equal to 0 (0);
 */
function arraySign (nums) {
  let product = 1;
  for (let i = 0; i < nums.length; i++) {
    product *= nums[i];
  }
  let res;
  if (product > 0) {
    res = 1;
  } else if (product < 0) {
    res = -1;
  } else {
    res = 0;
  }
  return res;
}

File: src/algorithms/array-sign.py
def array_sign(nums):
    """You are given an integer array nums where you must find the product x of all values in the array nums. Return 1 if x is positive, -1 if x 
    is negative, and 0 if x is equal to 0.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns 1 if the product is positive, -1 if it is negative, or 0
        if it is equal to 0.
    """
    sign = 1
    for num in nums:
        if num == 0:
            return 0
        elif num < 0:
            sign += 1
    if sign % 2 == 0:
        return -1
    else:
        return 1
